core/templates/pages/contributor-dashboard-page/contributions-and-review/contributions-and-review.component.ts

```diff
@@ -13,543 +13,641 @@
-              return skillRubrics;
-            },
-            skillDifficulty: function() {
-              return skillDifficulty;
-            },
-            suggestionId: function() {
-              return suggestionId;
-            },
-            editSuggestionCallback: () => openQuestionSuggestionModal
-          },
-          controller: 'QuestionSuggestionReviewModalController'
-        }).result.then(function(result) {
-          ContributionAndReviewService.reviewSkillSuggestion(
-            targetId, suggestionId, result.action, result.reviewMessage,
-            result.skillDifficulty, resolveSuggestionSuccess, () => {
-              AlertsService.addInfoMessage('Failed to submit suggestion.');
-            });
-        }, function() {
-          // Note to developers:
-          // This callback is triggered when the Cancel button is clicked.
-          // No further action is needed.
-          $rootScope.$applyAsync();
-        });
+import { Rubric } from 'domain/skill/rubric.model';
+import { SkillBackendApiService } from 'domain/skill/skill-backend-api.service';
+import { MisconceptionSkillMap } from 'domain/skill/MisconceptionObjectFactory';
+import { Question, QuestionBackendDict, QuestionObjectFactory } from 'domain/question/QuestionObjectFactory';
+import { ActiveContributionDict, TranslationSuggestionReviewModalComponent } from '../modal-templates/translation-suggestion-review-modal.component';
+import { ContributorDashboardConstants } from 'pages/contributor-dashboard-page/contributor-dashboard-page.constants';
+import { QuestionSuggestionReviewModalComponent } from '../modal-templates/question-suggestion-review-modal.component';
+import { TranslationTopicService } from 'pages/exploration-editor-page/translation-tab/services/translation-topic.service';
+import { FormatRtePreviewPipe } from 'filters/format-rte-preview.pipe';
+import { UserService } from 'services/user.service';
+import { AlertsService } from 'services/alerts.service';
+import { ContextService } from 'services/context.service';
+import { ContributionAndReviewService } from '../services/contribution-and-review.service';
+import { ContributionOpportunitiesService } from '../services/contribution-opportunities.service';
+
+export interface Suggestion {
+  change: {
+    skill_id?: string;
+    content_html: string;
+    translation_html: string | string[];
+    question_dict?: QuestionBackendDict;
+    skill_difficulty?: string[];
+  };
+  status: string;
+  suggestion_type: string;
+  target_id: string;
+  suggestion_id: string;
+  author_name?: string;
+}
+
+export interface QuestionContributionsSummaryList {
+  id: string;
+  heading: string;
+  subheading: string;
+  labelText: string;
+  labelColor: string;
+  actionButtonTitle: string;
+}
+
+export interface TranslationContributionsSummaryList {
+  id: string;
+  heading: string;
+  subheading: string;
+  labelText: string;
+  labelColor: string;
+  actionButtonTitle: string;
+}
```

Comment:
Shouldn't these types be declared as their individual item equivalents and then instantiated as an array, e.g. `QuestionContributionsSummary[]`?

Reply:
updated PTAL